frontend/src/components/TE/Versions.vue

```diff
@@ -0,0 +1,33 @@
+<template>
+  <div>
+    <article v-for="(version, index) in versions" :key="index">
+      <h3>Version: {{ index + 1 }}</h3>
+      <p>
+        {{ formatDurationUtc(version.timestamp, "MM-DD-YYYY hh:mm a", "ms") }}
```

Comment:
Either the function is called incorrectly or it is used incorrectly. Here we render a timestamp, not a duration. And not in UTC but in the local timezone. Please reconsider.